### ISSUE TITLE
Implement the quiz timer

### ISSUE THREAD

**Issue by GeorgZs** [DIT113-V22/group-01#41]
As a user, I want to be able to have a quiz timer, in order to have the real exam experience.

**Acceptance Criteria:**
- [x] The time left is shown on the quiz screen
- [x] The time is for the whole quiz, not a single question
- [x] The timer is only started when enabled in the quiz options
- [x] When the last question is answered the timer is stopped
- [x] The time elapsed is saved at the results and shown on the result screen
- [x] If the timer finishes before all questions are answered a popup is shown
  - [x] The popup explains the time is up
  - [x] The popup has a button to go to the results
  - [x] The unanswered questions are added as wrong and no answer index is set
- [x] This behavior is tested using automated unit tests

**Comment by IVidackovic** [comment]
Further work will be done once the timer implementation is merged into issue #37, as the UI as well as the state classes are required for the fulfilment of all acceptance criteria
